src/main/groovy/nextflow/executor/AwsBatchExecutor.groovy

```diff
@@ -318,13 +318,18 @@ class AwsBatchFileCopyStrategy extends SimpleFileCopyStrategy {
         """
         # aws helper
         nxf_s3_upload() {
-          local name=\$1
-          local s3path=\$2
-          if [[ -d \$name ]]; then
-            aws s3 cp \$name \$s3path --quiet --storage-class $S3StorageClass --recursive $S3EncryptionCommand
-          else
-            aws s3 cp \$name \$s3path --quiet --storage-class $S3StorageClass $S3EncryptionCommand
-          fi
+            local s3path="\${!#}"
+            local length=\$((\$#-1))
+            local files=\${@:1:\$length}
+            for f in \$files;do
+              if [[ -d \$f ]]; then
+                echo "aws s3 cp \$f \$s3path/\$f --quiet --storage-class $S3StorageClass --recursive $S3EncryptionCommand"
+                aws s3 cp \$f \$s3path/\$f --quiet --storage-class $S3StorageClass --recursive $S3EncryptionCommand
+              else
+                echo "aws s3 cp \$f \$s3path/\$f --quiet --storage-class $S3StorageClass $S3EncryptionCommand"
+                aws s3 cp \$f \$s3path/\$f --quiet --storage-class $S3StorageClass $S3EncryptionCommand
+              fi
+          done
         }
```

Comment:
I think it could be possible to simplify a bit do not expanding the wildcards when invoking  `nxf_s3_upload` just wrapping the first parameter in single quotes. Then it would become: 

```
 	nxf_s3_upload() {
	  local pattern=\$1
	  local s3path=\$2
	  for name in $patten; do 
		  if [[ -d \$name ]]; then
			aws s3 cp \$name \$s3path --quiet --storage-class $S3StorageClass --recursive $S3EncryptionCommand
		  else
			aws s3 cp \$name \$s3path --quiet --storage-class $S3StorageClass $S3EncryptionCommand
		  fi
	  done 
	}
```